### PR TITLE
feat: list objects & buckets

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

support list objects by id and list buckets by id

### Rationale

N/A

### Example

```
curl --location --request POST 'localhost:9133/?objects-query' \
--header 'Authorization: authTypeV2 ECDSA-secp256k1, Signature=1234567812345678123456781234567812345678123456781234567812345678' \
--header 'Content-Type: application/json' \
--data-raw '{
    "ids": [333,666]
}'
```

```
curl --location --request POST 'localhost:9133/?buckets-query' \
--header 'Authorization: authTypeV2 ECDSA-secp256k1, Signature=1234567812345678123456781234567812345678123456781234567812345678' \
--header 'Content-Type: application/json' \
--data-raw '{
    "ids": [222]
}'
```

### Changes

Notable changes: 
* support list objects by id and list buckets by id

### CODE REVIEW

File: modular/gater/metadata_handler.go [hunk 1]
```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
...
+	reqCtx, _ = NewRequestContext(r, g)
+
+	err = json.NewDecoder(r.Body).Decode(&objectIDs)
+	if err != nil {
+		log.Errorf("failed to parse object ids", "error", err)
+		err = ErrInvalidQuery
+		return
+	}
+
+	if len(objectIDs.IDs) == 0 {
```
Comment (ruojunm) [inline modular/gater/metadata_handler.go:538]: How many input IDs shall we allow at most?  Can we have a limit?

File: modular/gater/metadata_handler.go [hunk 2]
```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
...
+	reqCtx, _ = NewRequestContext(r, g)
+
+	err = json.NewDecoder(r.Body).Decode(&bucketIDs)
+	if err != nil {
+		log.Errorf("failed to parse bucket ids", "error", err)
+		err = ErrInvalidQuery
+		return
+	}
+
+	if len(bucketIDs.IDs) == 0 {
```
Comment (ruojunm) [inline modular/gater/metadata_handler.go:590]: ditto

File: modular/gater/metadata_handler.go (hunk 1 above)
Comment (BarryTong65) [reply modular/gater/metadata_handler.go]: It has been fixed, and is currently set to a limit of 1000

File: modular/gater/metadata_handler.go (hunk 2 above)
Comment (BarryTong65) [reply modular/gater/metadata_handler.go]: fixed

**Review by will-2012 — APPROVED** [review]

**Review by ruojunm — APPROVED** [review]

**Pull request merged**